src/libs/ajax/User.ts

```diff
@@ -0,0 +1,359 @@
+import _ from 'lodash/fp';
+import * as qs from 'qs';
+import {
+  authOpts,
+  fetchBond,
+  fetchEcm,
+  fetchOrchestration,
+  fetchRex,
+  fetchSam,
+  jsonBody,
+} from 'src/libs/ajax/ajax-common';
+import { getConfig } from 'src/libs/config';
+import { getTerraUser, TerraUserProfile } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+
+export interface SamUserRegistrationStatusResponse {
+  userSubjectId: string;
+  userEmail: string;
+  enabled: boolean;
+}
+
+export interface SamUserTosStatusResponse {
+  userInfo: {
+    userSubjectId: string;
+    userEmail: string;
+  };
+  enabled: {
+    ldap: boolean;
+    allUsersGroup: boolean;
+    google: boolean;
+  };
+}
+
+export interface SamUserTosComplianceStatusResponse {
+  userId: string;
+  userHasAcceptedLatestTos: boolean;
+  permitsSystemUsage: boolean;
+}
+
+export type TerraUserPreferences = {
+  starredWorkspaces?: string;
+} & {};
```

Comment:
What is the `& {}` for here?

Reply:
From the changes in state.ts, I'm guessing this is the type for other arbitrary preferences?

If that is the case, then maybe something like this would be clearer?

```suggestion
} & {
  [key: string]?: string;
};
```

Reply:
That makes more sense, I was not sure the best way to represent that 👍